src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -641,12 +657,7 @@ protected SymbolAndDiagnostics FindImplementationForInterfaceMemberWithDiagnosti
                         return SymbolAndDiagnostics.Empty;
                     }
 
-                    if (info.implementationForInterfaceMemberMap == null)
-                    {
-                        Interlocked.CompareExchange(ref info.implementationForInterfaceMemberMap, new ConcurrentDictionary<Symbol, SymbolAndDiagnostics>(), null);
-                    }
-
-                    return info.implementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
+                    return info.ImplementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
```

Comment:
No, we didn't implement it. So, yes, there's a Func allocation. I can eliminate it by doing a TryGet followed by a TryAdd (which is essentially what GetOrAdd does anyway)